azurerm/internal/services/hdinsight/parse/hd_insight_cluster.go

```diff
@@ -0,0 +1,69 @@
+package parse
+
+// NOTE: this file is generated via 'go:generate' - manual changes will be overwritten
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type HDInsightClusterId struct {
+	SubscriptionId string
+	ResourceGroup  string
+	ClusterName    string
+}
+
+func NewHDInsightClusterID(subscriptionId, resourceGroup, clusterName string) HDInsightClusterId {
+	return HDInsightClusterId{
+		SubscriptionId: subscriptionId,
+		ResourceGroup:  resourceGroup,
+		ClusterName:    clusterName,
+	}
+}
+
+func (id HDInsightClusterId) String() string {
+	segments := []string{
+		fmt.Sprintf("Cluster Name %q", id.ClusterName),
+		fmt.Sprintf("Resource Group %q", id.ResourceGroup),
+	}
+	segmentsStr := strings.Join(segments, " / ")
+	return fmt.Sprintf("%s: (%s)", "H D Insight Cluster", segmentsStr)
```

Comment:
that's a generator bug we should probably fix 🤔

Reply:
"Fixed" by renaming 👀